src/NServiceBus.AcceptanceTests/ConfigureEndpointAcceptanceTestingTransport.cs

```diff
@@ -46,16 +46,13 @@ public Task Configure(string endpointName, EndpointConfiguration configuration,
             .UseNativePubSub(useNativePubSub)
             .UseNativeDelayedDelivery(useNativeDelayedDelivery);
 
-        if (!useNativePubSub)
```

Comment:
Is there any downside to having this enabled all the time?

Reply:
the problem was with the test which emulates a subscriber which has been moved to the new version of a transport which supports native pubsub but is expected to also send message driven subscriptions due to the migration mode. Specifically this endpoint setup here: https://github.com/Particular/NServiceBus/pull/5413/files#diff-d3ea5ed90372e02637c61796b8f96271R53
We're using the `RegisterPublisher` API provided by the acceptance testing framework to register the mandatory routes for the subscription messages. This is currently the only easy way to configure the publisher routes for message driven pub/sub in acceptance tests. With that if check, thouse routes wouldn't be registered however and the test will fail because of missing routes.
The registered routes are only accessed by the `MessageDrivenSubscriptions` (and the migration mode) feature and having them registered by default doesn't have any impact on native pub/sub tests.